tests/test_base_availability_zone_manager.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common.h"
+#include "mock_file_ops.h"
+#include "mock_json_utils.h"
+#include "mock_logger.h"
+
+#include <multipass/base_availability_zone_manager.h>
+#include <multipass/constants.h>
+#include <multipass/exceptions/availability_zone_exceptions.h>
+
+#include <QString>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+namespace mpt = multipass::test;
+using namespace testing;
+
+struct TestBaseAvailabilityZoneManager : public Test
+{
+    TestBaseAvailabilityZoneManager() : mock_logger{mpt::MockLogger::inject()}
```

Comment:
Same here, you can initialize `mock_logger` where it's defined, no need to put it in the initializer list.